File: CloudSystem/src/Cloud/network/protocol/packet/RegisterServerPacket.php
<?php

namespace Cloud\network\protocol\packet;

class RegisterServerPacket extends Packet {

    public string $name = "";
    private int $port = 0;

    public function encode(): void {
        parent::encode();
        $this->put($this->name);
        $this->put($this->port);
    }

    public function decode(): void {
        parent::decode();
        $this->name = $this->get();
        $this->port = $this->get();
    }

    public function getId(): int {
        return self::ID_REGISTER_SERVER;
    }

    public static function create(string $name, int $port): self {
        $pk = new self;
        $pk->name = $name;
        $pk->port = $port;
        return $pk;
    }
}

File: CloudBridge-WD/src/main/java/de/r3pt1s/CloudBridge/network/protocol/packet/RegisterServerPacket.java
// This class was created by r3pt1s
package de.r3pt1s.CloudBridge.network.protocol.packet;

public class RegisterServerPacket extends Packet {

    public Object name = "";
    public Object port = 0;

    @Override
    public void encode() {
        super.encode();
        put(name);
        put(port);
    }

    @Override
    public void decode() {
        super.decode();
        name = get();
        port = get();
    }

    @Override
    public Integer getId() {
        return ID_REGISTER_SERVER;
    }

    public static RegisterServerPacket create(String name, int port) {
        RegisterServerPacket pk = new RegisterServerPacket();
        pk.name = name;
        pk.port = port;
        return pk;
    }
}
